### PR TITLE
added Developer documentation: how to contribute and some required st…

### PR DESCRIPTION
**Pull request by MarshmallowClive**
…eps to join the maintenance activity is explained(IBRAHIM-201732120159-Jifry-201732120165-Yussuf-201732120173-Clive-201732120173)

### CODE REVIEW

File: Documents/DevDoc.md [hunk 1]
```diff
@@ -1 +1,61 @@
-# DevDoc
+# Welcome to OAPS-Jifry
+
+**OAPS-Jifry** is a re-implementation of [OAPS](https://github.com/youmowang/zjnuservice) by Yun Dabang, and Ni Zhongjie, as OOAD Fall 2018 course project, using [JSP](https://en.wikipedia.org/wiki/JavaServer_Pages) technology.
+
+
+# How to contribute
+
+Currently, this project is maintained by team **MIJC** (Mahdi,IBRAHIM,Jifry, and Clive) and everyone else is welcomed to join us and help us to improve the project by any means.
```
Comment (spm2020spring) [inline Documents/DevDoc.md]: @MarshmallowClive 

`is welcomed` or `is welcome`?

Hui

File: Documents/DevDoc.md [hunk 2]
```diff
@@ -1 +1,61 @@
...
+
+Then, add your changes to the master repository, 
+
+    $git add .
+commit the changes, 
+
+    $git commit -m "your commit message"
+   and push them to the master repository.
+
+    $git push origin master
```
Comment (spm2020spring) [inline Documents/DevDoc.md:29]: @MarshmallowClive 

I think you don't want people to change the central master branch directly.

If you let contributors to do that, you do not understand [the feature branching workflow](https://github.com/spm2020spring/TeamCollaborationTutorial/blob/master/team.rst) very well.  


**Review-and-revise process** is an important practice I want to emphasize in this course.


**Hui**

ReviewRevise

File: Documents/DevDoc.md [hunk 3]
```diff
@@ -1 +1,61 @@
...
+## Development Plan with KanBan
```
Comment (spm2020spring) [inline Documents/DevDoc.md:30]: @MarshmallowClive 

This section is great.

Hui

File: Documents/DevDoc.md [hunk 4]
```diff
@@ -1 +1,61 @@
...
+For more information about the project's maintenance progress and to get a visual representation of the whole plan, see  [the project board on Kanboard](http://118.25.96.118/kanboard/?controller=BoardViewController&action=readonly&token=80ecf8f431a8dfafbc437dbc621b98d1c37c02790fd89f9cedc49a4c8d36).
+If you would like to join us on KanBoard, contact (lanhui at zjnu.edu.cn) to create an account for you, then contact (1525200991 at qq.com) to invite you to our project borad.
+## Which Area in the Project to Contribute to
+
+If you wish to contribute to the source code see section **"Project Environment"**.
+However, if you wish to contribute to the project's documentation, homepage, etc, of the non-source-code-related areas. You can skip **"Project Environment"** section.
```
Comment (spm2020spring) [inline Documents/DevDoc.md]: @MarshmallowClive 

Use the following instead:

if you wish to contribute to non-source-code-related areas, e.g.,   documentation,  homepage, you can ...
Hui

File: Documents/DevDoc.md [hunk 5]
```diff
@@ -1 +1,61 @@
...
+
+## Project Environment
+
+This project was developed using JSP and a number of other libraries and components. In order to run the project on your machine; you'll need to setup the appropriate environment.
```
Comment (spm2020spring) [inline Documents/DevDoc.md]: @MarshmallowClive 

`on your machine,`

Hui

File: Documents/DevDoc.md [hunk 6]
```diff
@@ -1 +1,61 @@
...
+The following is a list of the current running environment:
```
Comment (spm2020spring) [inline Documents/DevDoc.md]: @MarshmallowClive 

a list of tools or libraries you need to install:

Hui

File: Documents/DevDoc.md [hunk 7]
```diff
@@ -1 +1,61 @@
...
+* [Eclipse Jee 2019-09 IDE](https://www.eclipse.org/downloads/packages/release/2019-09/r)
+* [Apache Tomcat v7.0](https://tomcat.apache.org/download-70.cgi)
+* [JDK-13](https://www.oracle.com/java/technologies/javase-jdk13-downloads.html)
+* [Jstl-1.2](https://www.oracle.com/technetwork/java/index-137889.html)
+* [mysql-connector-java-8.0.12](https://dev.mysql.com/downloads/connector/j/)
+* [commons-fileupload-1.3](http://commons.apache.org/proper/commons-fileupload/download_fileupload.cgi)
+* [commons-io-2.4](https://commons.apache.org/proper/commons-io/download_io.cgi)
+
+**NOTE**: This is the current running system environment, there might be newer versions of any component, and you might use newer versions if you like.
+Also we would like to assume that you know how to setup the aforementioned components. If not, you can look it up on the internet.
```
Comment (spm2020spring) [inline Documents/DevDoc.md]: @MarshmallowClive 

"If not, you can look it up on the internet." Delete this sentence.  Most people know that without even being reminded.

Hui

File: Documents/DevDoc.md [hunk 8]
```diff
@@ -1 +1,61 @@
...
+
+After you have finished setting up the environment, and everything is up and running, see the following section to understand our contribution policy.
```
Comment (spm2020spring) [inline Documents/DevDoc.md]: @MarshmallowClive 

Replace "contribution policy" with "workflow".

Hui

File: Documents/DevDoc.md [hunk 9]
```diff
@@ -1 +1,61 @@
...
+## Follow the Feature-branching Workflow
+
+In this collaborative project development, we are trying to protect the central repository from unreasonable changes that may break the whole system, or may cause inconsistency in the repository. Hence, we follow the feature-branching workflow (please, see this section *[The feature-branching workflow](https://github.com/spm2020spring/TeamCollaborationTutorial/blob/master/team.rst)* for a tutorial on how to get started with this technique).
```
Comment (spm2020spring) [inline Documents/DevDoc.md]: @MarshmallowClive 

Delete "with this technique".

Hui

File: Documents/DevDoc.md [hunk 10]
```diff
@@ -1 +1,61 @@
...
+Simply, any change you may would like to do must be done on a separate branch, so that the change you made is being reviewed by a number of contributors to spot any problems or point some bugs in the change it self before it gets approved.
```
Comment (spm2020spring) [inline Documents/DevDoc.md]: @MarshmallowClive 

Replace "is being" with "is".

Replace "point" with "pinpoint".

Replace "it self" with "itself".

Hui

**Comment by spm2020spring** [comment]
@MarshmallowClive 

Thanks for the writing.

However, the content looks more like for **developer guidelines** than for **developer documentation**.  The developer documentation is supposed to help people understand the code.

Check the following two links and compare their difference:

- Developer guidelines: https://producingoss.com/en/getting-started.html#developer-guidelines 

- Developer documentation: https://producingoss.com/en/getting-started.html#developer-documentation

But don't worry. 

**Hui**

**Comment by MarshmallowClive** [comment]
Is welcome

Sent from Yahoo Mail on Android 
 
  On Sun, 10 May 2020 at 22:50, Software Project Management<notifications@github.com> wrote:   


@spm2020spring commented on this pull request.

In Documents/DevDoc.md:
> @@ -1 +1,61 @@
-# DevDoc
+# Welcome to OAPS-Jifry
+
+**OAPS-Jifry** is a re-implementation of [OAPS](https://github.com/youmowang/zjnuservice) by Yun Dabang, and Ni Zhongjie, as OOAD Fall 2018 course project, using [JSP](https://en.wikipedia.org/wiki/JavaServer_Pages) technology.
+
+
+# How to contribute
+
+Currently, this project is maintained by team **MIJC** (Mahdi,IBRAHIM,Jifry, and Clive) and everyone else is welcomed to join us and help us to improve the project by any means.

@MarshmallowClive

is welcomed or is welcome?

Hui

File: Documents/DevDoc.md (hunk 2 above)
Comment (hema-001) [reply Documents/DevDoc.md:29]: We did mention that in "Follow the feature-branching workflow" section.

What we meant in this part that he/she can directly append their names on the list of contributors.

Comment (spm2020spring) [reply Documents/DevDoc.md:29]: > We did mention that in "Follow the feature-branching workflow" section.
> 
> What we meant in this part that he/she can directly append their names on the list of contributors.

@hema-001 
@TanakaMichelle 

Thanks. Now I understand.   

Hui

**Review by hema-001 — APPROVED** [review]
All good.

**Pull request merged**